hyperspy/_signals/eds_tem.py

```diff
@@ -483,3 +527,66 @@ def create_model(self, auto_background=True, auto_add_lines=True,
+        methods of quantification.
+
+        Inputgiven by i*t*N, i the current, t the
+        acquisition time, and N the number of electron by unit electric charge.
+
+        Parameters
+        ----------
+        method : Defined as either 'zfactor' or 'cross_section'
+        If 'zfactor', the dose is given by i*t*N,
+        i is the current
+        t is the dwell time
+        N is the number of electrons by unit charge or 1/e.
+        If 'cross_section', the dose per unit is area is returned by i*t*N/A
+        A is the beam area or pixel area in nm^2
+        beam_current: Defined in nA with the symbol i above.
+        real_time: Defined in s with the symbol t above.
+        area: Defined in nm^2 with the symbol A above, is the illumination area
+        of the electron beam. If none is given it is extracted from the scale
+        axes_manager. Therefore we assume the probe is oversampling such that
+        the illumination area can be approximated to the pixel area of the
+        pectrum image.
+
+        Returns
+        --------
+        Dose in electrons (zfactor) or electrons per nm^2 (cross_section).
+        """
+
+        parameters = self.metadata.Acquisition_instrument.TEM
+        if beam_current == 'auto':
+            if 'beam_current' in self.metadata.Acquisition_instrument.TEM == False:
+                raise Exception ('It is not possible to carry out EDX \
+quantification_cross_section without a beam_current please set one using \
+self.metadata.Acquisition_instrument.TEM.beam_current and run quantification \
+again.')
+            else:
+                beam_current = parameters.beam_current
+        if real_time == 'auto':
+            real_time = parameters.Detector.EDS.real_time
+            if 'real_time' in self.metadata.Acquisition_instrument.TEM.Detector.EDS == False:
+                raise Exception ('Please note that your real time is set to\
+the default value of 0.5s. The function will still run. However, if this is \
+incorrect you should consider changing it using \
+self.metadata.Acquisition_instrument.TEM.Detector.EDS.real_time .')
+        if method == 'cross_section':
+            if area == 'auto':
+                pixel1 = self.axes_manager[0].scale
+                pixel2 = self.axes_manager[1].scale
+            if self.axes_manager[0].scale == 1 or self.axes_manager[1].scale == 1:
+                    warnings.warn('Please note your pixel width is set to the \
+default value of 1nm. The function will still run. However, if this is \
+incorrect you should consider changing this using the axes_manager.gui() or \
+axes_manger[0].scale functions.')
+            area = pixel1 * pixel2
+            return (real_time * beam_current * 1e-9) /(constants.e * area)
+        elif method == 'zeta':
+            return real_time * beam_current * 1e-9 / constants.e
+        else:
+            raise Error('no method provided')
```

Comment:
Although if I make the adjustment above so that @dnjohnstone suggested about making the method argument without a default. It will presumable error out before this point, so this error is no longer needed?